vm/devices/net/net_mana/src/lib.rs

```diff
@@ -749,6 +753,54 @@ impl<T: DeviceBacking> ManaQueue<T> {
             );
         }
     }
+
+    fn trace_tx_wqe_from_offset(&mut self, wqe_offset: u32) {
+        let size = size_of::<Wqe>(); // WQE is 512 bytes
```

Comment:
I still think its better to just read the WQEHeader and the needed OOB bytes to log the data below. WQE can be of varying size and so its not given that it will be always 512 bytes

Reply:
Wqe is always exactly 512. The oob_len can vary, but when I call wqe.oob() it figures that out and returns the exact slice I need.